### PR TITLE
Hair - code fixes based on the CR remarks for the Hair merge to Dev

### PR DESCRIPTION
**Pull request by Adi-Amazon**
- Changing Hot reload to be in the following frame + removing loop + bullet proof parallel async operation
- Naming refactors based on CR suggestions
- Improved and fixed comments
- Redundant files / interface removal
- Conflicts solved: Gems/AtomTressFX/gem.json

### CODE REVIEW

File: Gems/AtomTressFX/Assets/Shaders/HairLightTypes.azsli [hunk 1]
```diff
@@ -180,6 +180,11 @@ void ApplySimplePointLight(ViewSrg::SimplePointLight light, Surface surface, ino
     }
 }
 
+// The following function is currently not used and is written here for future testing.
+// The culling scheme requires some more preperation and testing in HairLighting.azsli
+// and once done, all light types can use the culling. 
+// Not that the current culling scheme assumes light types lists order and is bogus if
```
Comment (Adi-Amazon) [inline Gems/AtomTressFX/Assets/Shaders/HairLightTypes.azsli:186]: missing e: Not -> Note

File: Gems/AtomTressFX/Assets/Shaders/HairLightTypes.azsli [hunk 2]
```diff
@@ -578,8 +439,8 @@ float3 ApplyIblSpecular(Surface surface, LightingData lightingData)
 void ApplyIBL(Surface surface, inout LightingData lightingData)
 {
 //    float3 normal = normalize(float3(surface.tangent.z, -surface.tangent.x, surface.tangent.y));
-//    SetNormal(surface.tangent, normal, surface, lightingData);
-    SetNormal(surface.tangent, lightingData.dirToCamera, surface, lightingData);
+//    SetNormalAndUpdateLightingParams(surface.tangent, normal, surface, lightingData);
```
Comment (Adi-Amazon) [inline Gems/AtomTressFX/Assets/Shaders/HairLightTypes.azsli:442]: IBL is still WIP and this part will change in the coming weeks

**Review by amzn-tommy — APPROVED** [review]
A couple minor suggestions about the hot-reload comments, but otherwise looks good.

File: Gems/AtomTressFX/Code/Passes/HairSkinningComputePass.cpp [hunk 3]
```diff
@@ -172,34 +168,43 @@ namespace AZ
...
             {
-                if (!hairObject)
-                {
-                    return;
-                }
+                // The following mutex is used for blocking the shader switch when a hot load occurs, hence ensuring
+                // the shader exists and the same shader, data and dispatch items are used across all hair objects
+                // during this frame.
+                // Two cases exists:
+                // 1. Hot reload will be revoked first - if this mutex was blocking, the flag for hot reload will
```
[the remark this answers is not in the document]
Comment (Adi-Amazon) [reply Gems/AtomTressFX/Code/Passes/HairSkinningComputePass.cpp:177]: +1 tnx

**Review by antonmic — APPROVED** [review]

File: Gems/AtomTressFX/Code/Passes/HairPPLLRasterPass.h [hunk 4]
```diff
@@ -55,6 +55,7 @@ namespace AZ
                 static RPI::Ptr<HairPPLLRasterPass> Create(const RPI::PassDescriptor& descriptor);
 
                 bool AddDrawPacket(HairRenderObject* hairObject);
```
[the remark this answers is not in the document]
Comment (Adi-Amazon) [reply Gems/AtomTressFX/Code/Passes/HairPPLLRasterPass.h:57]: indeed - that one slipped and AR would have spanked me :)
Removed.

**Pull request merged**